Explorer/Assets/DCL/Landscape/Jobs/GenerateTreeInstancesJob.cs

```diff
@@ -0,0 +1,228 @@
+﻿using DCL.Landscape.Config;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Jobs;
+using Unity.Mathematics;
+using UnityEngine;
+using Random = Unity.Mathematics.Random;
+
+namespace DCL.Landscape.Jobs
+{
+    /// <summary>
+    /// This job gathers the treeNoise result from other job and does additional checks like:
+    ///     - Is there a neighbour owned parcel that overlaps with my asset radius?
+    ///     - Is there another asset of the same type within my radius?
+    /// </summary>
+
+    // TODO: Convert NativeHashMap into NativeArray's so we can use IJobParallelFor
+    [BurstCompile]
+    public struct GenerateTreeInstancesJob : IJob
```

Comment:
Just like you mentioned you can use `NativeArray` to convert to `IJobParallelFor`. Or you can use `NativeParallelHashMap`. 
But I don't think you really need a hashmap. Moreover with a native array you can preallocate a managed one, convert it to the native one and avoid additional allocation on the caller side